plugins/flytekit-pydantic-v2/flytekitplugins/pydantic/__init__.py

```diff
@@ -0,0 +1,25 @@
+from flytekit.types.directory import FlyteDirectory
+from flytekit.types.file import FlyteFile
+from flytekit.types.schema import FlyteSchema
+from flytekit.types.structured import StructuredDataset
+
+from . import transformer
+from .custom import (
+    deserialize_flyte_dir,
+    deserialize_flyte_file,
+    deserialize_flyte_schema,
+    deserialize_structured_dataset,
+    serialize_flyte_dir,
+    serialize_flyte_file,
+    serialize_flyte_schema,
+    serialize_structured_dataset,
+)
+
+setattr(FlyteFile, "serialize_flyte_file", serialize_flyte_file)
```

Comment:
🙈 i assume we're okay with this?  Any other approach? @pingsutw @eapolinario @kumare3 - what's another approach?  setters for these functions?

Reply:
I think this is the best way since we can't write the code like the given example in official pydantic documentation.
https://docs.pydantic.dev/latest/api/functional_validators/#pydantic.functional_validators.model_validator